### PR TITLE
Add vls128 support for ros branch

### PR DESCRIPTION
**Pull request by C-NR**
I added the support for the Velodyne VLS-128 to the ROS 2 branch. I took and modified the according code lines from the ROS 1 master branch at  tag 'version 1.7.0'. Its function has been tested on recorded data and live using a VLS-128 @DCMLR, Freie Universität Berlin.

### CODE REVIEW

File: velodyne_pointcloud/include/velodyne_pointcloud/rawdata.hpp [hunk 1]
```diff
@@ -121,8 +139,31 @@ struct raw_packet
 class RawData final
 {
 public:
+  // RawData();
```
Comment (JWhitleyWork) [inline velodyne_pointcloud/include/velodyne_pointcloud/rawdata.hpp:142]: Please remove commented-out code.

File: velodyne_pointcloud/include/velodyne_pointcloud/rawdata.hpp [hunk 2]
```diff
@@ -121,8 +139,31 @@ struct raw_packet
...
+   * @param calibration_file path to the calibration file
+   * @param model sensor model type
+   * @param max_range_ cutoff for maximum range
+   * @param min_range_ cutoff for minimum range
+   * @returns 0 if successful;
+   *           errno value for failure
+   */
+  // int setupOffline(std::string calibration_file, std::string model, double max_range_,
+  //                  double min_range_);
+
```
Comment (JWhitleyWork) [inline velodyne_pointcloud/include/velodyne_pointcloud/rawdata.hpp:166]: Please remove commented-out code.

File: velodyne_pointcloud/include/velodyne_pointcloud/rawdata.hpp [hunk 3]
```diff
@@ -121,8 +139,31 @@ struct raw_packet
 class RawData final
 {
 public:
+  // RawData();
   RawData(const std::string & calibration_file, const std::string & model);
 
+  ~RawData()
+  {
+  }
```
Comment (JWhitleyWork) [inline velodyne_pointcloud/include/velodyne_pointcloud/rawdata.hpp:147]: Please remove this empty destructor. It disables copy/move constructors and isn't being used.

File: velodyne_pointcloud/include/velodyne_pointcloud/rawdata.hpp [hunk 4]
```diff
@@ -135,15 +176,20 @@ class RawData final
 
 private:
   /** configuration parameters */
-  struct Config
+  typedef struct
```
Comment (JWhitleyWork) [inline velodyne_pointcloud/include/velodyne_pointcloud/rawdata.hpp:179]: Why this change?

File: velodyne_pointcloud/include/velodyne_pointcloud/rawdata.hpp [hunk 5]
```diff
@@ -135,15 +176,20 @@ class RawData final
...
   {
     std::string model;
-    double min_range;             ///< minimum range to publish
+    std::string calibrationFile;  ///< calibration file name
```
Comment (JWhitleyWork) [inline velodyne_pointcloud/include/velodyne_pointcloud/rawdata.hpp:182]: Why change the ordering of this struct? And why is it that the path to the calibration file is needed here now when it was not before?

File: velodyne_pointcloud/src/lib/rawdata.cpp [hunk 6]
```diff
@@ -266,10 +324,6 @@ void RawData::unpack(
       tmp.bytes[0] = block.data[k];
       tmp.bytes[1] = block.data[k + 1];
 
-      if (tmp.bytes[0] == 0 && tmp.bytes[1] == 0) {  // no laser beam return
-        continue;
-      }
-
```
Comment (JWhitleyWork) [inline velodyne_pointcloud/src/lib/rawdata.cpp:272]: Why was this removed?

File: velodyne_pointcloud/src/lib/rawdata.cpp [hunk 7]
```diff
@@ -384,21 +450,164 @@ void RawData::unpack(
         intensity = (intensity < min_intensity) ? min_intensity : intensity;
         intensity = (intensity > max_intensity) ? max_intensity : intensity;
 
-        float time = 0;
-        if (timing_offsets_.size()) {
-          time = timing_offsets_[i][j] + time_diff_start_to_this_packet;
-        }
-
```
Comment (JWhitleyWork) [inline velodyne_pointcloud/src/lib/rawdata.cpp:391]: If this is all going to get skipped for the 128, why remove this code?

**Review by JWhitleyWork — CHANGES_REQUESTED** [review]
I have some concerns.

File: velodyne_pointcloud/src/lib/rawdata.cpp (hunk 7 above)
Comment (C-NR) [reply velodyne_pointcloud/src/lib/rawdata.cpp:391]: it is not removed it moved up to line 336 to 339, because of line 341 to 346. (also ee the comment there.

File: velodyne_pointcloud/src/lib/rawdata.cpp (hunk 6 above)
Comment (C-NR) [reply velodyne_pointcloud/src/lib/rawdata.cpp:272]: Got to check that, but it looks like a merge mistake. So i will revert that change

File: velodyne_pointcloud/include/velodyne_pointcloud/rawdata.hpp (hunk 5 above)
Comment (C-NR) [reply velodyne_pointcloud/include/velodyne_pointcloud/rawdata.hpp:182]: Only reason is that the ROS1 fork had that included here. I will double check if its used and if not remove here.

**Comment by JWhitleyWork** [comment]
@C-NR I know yours was submitted first, but would you instead mind testing out #493? Their PR is a little more complete (launch and config files, fewer changes, etc.).

**Comment by JWhitleyWork** [comment]
Closing after 5 months of no response.
